src/python/pants/backend/cc/lint/clangformat/rules.py

```diff
@@ -39,47 +40,39 @@ class ClangFormatRequest(FmtTargetsRequest):
 
 @rule(level=LogLevel.DEBUG)
 async def clangformat_fmt(request: ClangFormatRequest.Batch, clangformat: ClangFormat) -> FmtResult:
-    # Look for any/all of the clang-format configuration files (recurse sub-dirs)
-    config_files_get = Get(
-        ConfigFiles,
-        ConfigFilesRequest,
-        clangformat.config_request(request.snapshot.dirs),
-    )
-
-    clangformat_pex, config_files = await MultiGet(
-        Get(Pex, PexRequest, clangformat.to_pex_request()), config_files_get
+    clangformat_pex, config_files = await concurrently(
+        create_pex(clangformat.to_pex_request()),
+        find_config_file(clangformat.config_request(request.snapshot.dirs)),
     )
 
     # Merge source files, config files, and clang-format pex process
-    input_digest = await Get(
-        Digest,
+    input_digest = await merge_digests_request_to_digest(
         MergeDigests(
-            [
-                request.snapshot.digest,
-                config_files.snapshot.digest,
-                clangformat_pex.digest,
-            ]
-        ),
+            [request.snapshot.digest, config_files.snapshot.digest, clangformat_pex.digest]
+        )
     )
 
-    result = await Get(
-        ProcessResult,
-        PexProcess(
-            clangformat_pex,
-            argv=(
-                "--style=file",  # Look for .clang-format files
-                "--fallback-style=webkit",  # Use WebKit if there is no config file
-                "-i",  # In-place edits
-                "--Werror",  # Formatting warnings as errors
-                *clangformat.args,  # User-added arguments
-                *request.files,
-            ),
-            input_digest=input_digest,
-            output_files=request.files,
-            description=f"Run clang-format on {pluralize(len(request.files), 'file')}.",
-            level=LogLevel.DEBUG,
+    argv = (
+        "--style=file",  # Look for .clang-format files
+        "--fallback-style=webkit",  # Use WebKit if there is no config file
+        "-i",  # In-place edits
+        "--Werror",  # Formatting warnings as errors
+        *clangformat.args,  # User-added arguments
+        *request.files,
+    )
+    result = await fallible_to_exec_result_or_raise(
+        **implicitly(
```

Comment:
This solves a rule-graph error, where fallible_to... isn't expecting a PexProcess.
As per @stuhood  https://pantsbuild.slack.com/archives/C0D7TNJHL/p1718575366615099?thread_ts=1718115214.921379&cid=C0D7TNJHL

> if any sort of argument conversion is supposed to happen, then it should not be a positional argument: rather an **implicitly(PexProcess(..))
> you call the function whose result you want, but you implicitly convert the arguments